stores/metadata.go

```diff
@@ -644,6 +646,11 @@ func sqlConcat(db *gorm.DB, a, b string) string {
 	return fmt.Sprintf("CONCAT(%s, %s)", a, b)
 }
 
+func sqlHasPrefix(col, prefix string) clause.Expr {
+	query := fmt.Sprintf("%s LIKE ? AND SUBSTR(%s, 1, ?) = ?", col, col)
```

Comment:
I think it was proposed due to LIKE being slightly faster but I haven't benchmarked it tbh. 
If you want I can change it since it will definitely be cleaner.